controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -121,8 +135,30 @@ func (r *StorageConsumerUpgradeReconciler) Reconcile(ctx context.Context, reques
 		return reconcile.Result{}, err
 	}
 
-	if err = r.reconcileStorageConsumer(ctx, storageCluster, storageConsumer, consumerConfigMapName); err != nil {
-		return reconcile.Result{}, err
+	if internalConsumer {
+		newConsumer := &ocsv1alpha1.StorageConsumer{}
+		newConsumer.Name = defaults.LocalStorageConsumerName
+		newConsumer.Namespace = storageConsumer.Namespace
+		newConsumer.Spec.Enable = true
+		newConsumer.Spec.ResourceNameMappingConfigMap.Name = consumerConfigMapName
+		if err := r.Client.Create(ctx, newConsumer); err != nil && !errors.IsAlreadyExists(err) {
+			return reconcile.Result{}, err
+		}
+		if err = r.reconcileStorageClient(ctx, storageCluster.Name, string(newConsumer.UID)); err != nil {
+			return reconcile.Result{}, err
+		}
+		util.AddAnnotation(storageCluster, util.ProviderModeCompatibleInternalConsumerUID, string(storageConsumer.UID))
+		if err := r.Client.Update(ctx, storageCluster); err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to add storage sluster compatibility mode annotation: %w", err)
+		}
+		// Delete the old consumer
+		if err = r.Client.Delete(ctx, storageConsumer); err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		if err = r.reconcileStorageConsumer(ctx, storageCluster, storageConsumer, consumerConfigMapName); err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
We are already doing a create or update on the every consumer. why do we need a special path here? If you select the correct name there should be no 'other path' for creation of the new consumer